packages/calcite-components/src/components/input/input.scss

```diff
@@ -1,4 +1,48 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-input-corner-radius: defines the border radius of the component.
```

Comment:
Conditionally handle the closable button with this value: 
<img width="421" alt="Screenshot 2024-02-16 at 9 15 36 AM" src="https://github.com/Esri/calcite-design-system/assets/4733155/db834a2d-e491-44f4-b933-ea8d708b5e7a">

Reply:
I already have the styles there to prevent an example like you've shown above. The input will not add a rounded border when a ::slotted action is added. However, I can not style the button as shown above as the button is not currently styled in such a way as to allow that.

Reply:
Agreed, we shouldn't apply it to slotted content. But that clear button is rendered internally, IMO we should treat it like the number buttons which you already handle:

```
border-start-start-radius: var(--calcite-input-corner-radius);
border-end-start-radius: var(--calcite-input-corner-radius);
```  

It's only an issue when the close button is adjacent to the edge, you're already handling it in these cases, I think it just needs a conditional for the above state:
<img width="438" alt="Screenshot 2024-02-16 at 5 23 32 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/c8f26127-8e83-4644-a85e-2b4ab306bd4d">
<img width="326" alt="Screenshot 2024-02-16 at 5 22 30 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/194794e4-1849-4b78-bd09-0b6d2a6c8167">